retrieval/discovery/marathon/conversion.go

```diff
@@ -24,18 +22,25 @@ func AppsToTargetGroups(apps *AppList) map[string]*config.TargetGroup {
 func createTargetGroup(app *App) *config.TargetGroup {
 	var (
 		targets = targetsForApp(app)
-		source  = targetGroupName(app)
+		source  = sanitizeName(app.ID)
```

Comment:
Also removed other sanitizations as they end up in labels. Don't feel like we should alter input if not necessary.